pkg/controlplane/controller/defaultservicecidr/default_servicecidr_controller.go

```diff
@@ -175,18 +190,26 @@ func (c *Controller) syncStatus(serviceCIDR *networkingapiv1.ServiceCIDR) {
 
 	// This controller will set the Ready condition to true if the Ready condition
 	// does not exist and the CIDR values match this controller CIDR values.
+	sameConfig := reflect.DeepEqual(c.cidrs, serviceCIDR.Spec.CIDRs)
 	for _, condition := range serviceCIDR.Status.Conditions {
 		if condition.Type == networkingapiv1.ServiceCIDRConditionReady {
 			if condition.Status == metav1.ConditionTrue {
+				// ServiceCIDR is Ready and config matches this apiserver
+				// nothing else is required
+				if sameConfig {
+					return
+				}
+			} else if !c.reportedNotReadyCondition {
+				klog.InfoS("default ServiceCIDR condition Ready is not True, please validate your cluster's network configuration for this ServiceCIDR", "status", condition.Status, "reason", condition.Reason, "message", condition.Message)
+				c.eventRecorder.Eventf(serviceCIDR, v1.EventTypeWarning, condition.Reason, condition.Message)
+				c.reportedNotReadyCondition = true
 				return
 			}
-			klog.Infof("default ServiceCIDR condition Ready is not True: %v", condition.Status)
-			c.eventRecorder.Eventf(serviceCIDR, v1.EventTypeWarning, condition.Reason, condition.Message)
-			return
 		}
 	}
-	// set status to ready if the ServiceCIDR matches this configuration
-	if reflect.DeepEqual(c.cidrs, serviceCIDR.Spec.CIDRs) {
+	// Not condition set, set status to ready if the ServiceCIDR matches this configuration
```

Comment:
NIT: Am I misreading, or is this a typo?
```suggestion
	// No condition set, set status to ready if the ServiceCIDR matches this configuration
```
?

Reply:
typo